### DISCUSSION TITLE
Extend Error Notifications to REST API actions

### DISCUSSION THREAD

**Discussion by bturi** [apache/airflow#29774] [Ideas]
**Description**

Extend DAG's `email_on_failure` feature to the case when a Task/DAG state is modified via Airflow's REST API endpoints such as `updateTaskInstancesState`.

**Use case / motivation**
Suppose a Task state is set to 'failed' via the official REST API. It would be convenient if the `email_on_failure` setting of the respective DAG would also apply here.

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!
